core/domain/exp_services.py

```diff
@@ -830,6 +837,7 @@ def _create_exploration(
     # but the creation of an exploration object will fail.
     exploration.validate()
     rights_manager.create_new_exploration_rights(exploration.id, committer_id)
+    exploration = classifier_services.train(exploration)
```

Comment:
@anmolshkl . I went through the functions and found that _save_exploration() and _create_exploration() are somewhat mutually exclusive. Most of the functions that create new explorations uses the latter and then _save_exploration() isn't involved at all. For example, the save_new_exploration_from_yaml_and_assets() never calls _save_Exploration().